tools/util.wdl

```diff
@@ -417,3 +417,173 @@ task make_coverage_regions_beds {
         maxRetries: max_retries
     }
 }
+
+task global_phred_scores {
+    meta {
+        description: "Calculates statistics about PHRED scores of the input BAM."
+    }
+
+    input {
+        File bam
+        String prefix = basename(bam, ".bam")
+        Int memory_gb = 4
+        Int modify_disk_size_gb = 0
+        Int max_retries = 1
+    }
+
+    Float bam_size = size(bam, "GiB")
+    Int disk_size_gb = ceil(bam_size) + 10 + modify_disk_size_gb
+
+    String outfile_name = prefix + ".global_PHRED_scores.txt"
+
+    command <<<
+        set -euo pipefail
+
+        BAM="~{bam}" PREFIX="~{prefix}" python3 - <<END
+import os  # lint-check: ignore
+import pysam  # lint-check: ignore
+from collections import defaultdict  # lint-check: ignore
+
+bam_path = os.environ["BAM"]
+bam = pysam.AlignmentFile(bam_path, "rb")
+
+tot_quals = defaultdict(lambda: 0)
+mapped_quals = defaultdict(lambda: 0)
+unmapped_quals = defaultdict(lambda: 0)
+middle_tot_quals = defaultdict(lambda: 0)
+middle_mapped_quals = defaultdict(lambda: 0)
+middle_unmapped_quals = defaultdict(lambda: 0)
+for read in bam:
+    # only count primary alignments and unmapped reads
+    if read.is_secondary and not read.is_unmapped:
+        continue
+
+    cur_quals = read.query_alignment_qualities  # array of phred scores
+    for qual in cur_quals:
+        tot_quals[qual] += 1
+        if read.is_unmapped:
+            unmapped_quals[qual] += 1
+        else:
+            mapped_quals[qual] += 1
+
+    middle_pos = len(cur_quals) // 2  # middle base of read
+    middle_score = cur_quals[middle_pos]
+    middle_tot_quals[middle_score] += 1
+    if read.is_unmapped:
+        middle_unmapped_quals[middle_score] += 1
+    else:
+        middle_mapped_quals[middle_score] += 1
+
+prefix = os.environ["PREFIX"]
+outfile = open(prefix + ".global_PHRED_scores.txt", "w")
```

Comment:
Pretty small point, but should this have a `.tsv` ext instead of `.txt`? I think everywhere else we produce TSVs (mainly ngsderive) we use a `.txt` extension. I think I'd prefer to be more specific and change _all_ those to end in `.tsv`. But in the mean time, using `.txt` here is more consistent with the rest of our tasks.

I just see it as a matter of being more specific about the file contents is better. I don't see any reason (besides history) to stick with `.txt`.

Reply:
I prefer `.tsv` for tab-delimited data as well.